File: Sc.Models/Dtos/Suppliers/Junctions/SupplierOfferingSmartSpecializationDto.cs
using Sc.Models.Dtos.Nomenclatures.SmartSpecializations;
using Sc.Models.Entities.Base;
using Sc.Models.Enums.Suppliers;

namespace Sc.Models.Dtos.Suppliers.Junctions
{
    public class SupplierOfferingSmartSpecializationDto : Entity
    {
        public int SupplierOfferingId { get; set; }

        public int SmartSpecializationId { get; set; }
        public SmartSpecializationDto SmartSpecialization { get; set; }

        public SupplierOfferingSmartSpecializationType Type { get; set; }
    }
}

File: WebApplication/src/suppliers/dtos/junctions/supplier-offering-smart-specialization.dto.ts
import { SmartSpecializationDto } from "src/nomenclatures/dtos/smart-specializations/smart-specialization.dto";
import { SupplierOfferingSmartSpecializationType } from "src/suppliers/enums/supplier-offering-smart-specialization-type.enum";

export class SupplierOfferingSmartSpecializationDto {
    id: number;

    supplierOfferingId: number;

    smartSpecializationId: number;
    smartSpecialization: SmartSpecializationDto;

    type: SupplierOfferingSmartSpecializationType = SupplierOfferingSmartSpecializationType.secondary;
}